calendar/src/main/java/com/toggl/calendar/domain/CalendarItemsSelector.kt

```diff
@@ -0,0 +1,26 @@
+package com.toggl.calendar.domain
+
+import com.toggl.environment.services.calendar.CalendarEvent
+import com.toggl.models.domain.TimeEntry
+import org.threeten.bp.OffsetDateTime
+
+fun calendarItemsSelector(
+    calendarLayoutCalculator: CalendarLayoutCalculator,
+    timeEntries: Map<Long, TimeEntry>,
+    calendarEvents: List<CalendarEvent>,
+    date: OffsetDateTime
+): List<CalendarItem> {
+    val localDate = date.toLocalDate()
+    val filteredTimeEntries = timeEntries.filterValues {
+        val endTime = it.startTime + it.duration
+        it.startTime.toLocalDate() == localDate && (endTime == null || endTime.toLocalDate() == localDate)
+    }
+    val filteredEvents = calendarEvents.filter {
+        val endTime = it.startTime + it.duration
+        it.startTime.toLocalDate() == localDate && (endTime == null || endTime.toLocalDate() == localDate)
+    }
+    return filteredTimeEntries
+        .values.map { CalendarItem.ofTimeEntry(it) }
+        .plus(filteredEvents.map { CalendarItem.ofCalendarEvent(it) })
+        .run { calendarLayoutCalculator.calculateLayoutAttributes(this) }
```

Comment:
What do you think about using method references here?
```suggestion
    return filteredTimeEntries.values
        .map(CalendarItem::ofTimeEntry)
        .plus(filteredEvents.map(CalendarItem::ofCalendarEvent))
        .let(calendarLayoutCalculator::calculateLayoutAttributes)
```

Reply:
Yes, please

Reply:
Here's another example of why we don't need the companion method. This will not compile since you need to `CalendarItem.Companion::ofTimeEntry` while using the sealed class itself you'd simply `CalendarItem::TimeEntry`

Reply:
Actually, `CalendarItem::TimeEntry` doesn't work, unless I'm missing something (at least not on the version of kotlin we are using).

We can either keep the companion object method and use `map(CalendarItem::ofTimeEntry)`or use `map { CalendarItem.TimeEntry(it) } `.